### PR TITLE
Perf auto4

### PR DESCRIPTION
**Pull request by Paolob69**
This pull contains:
-  fix in the copy_pattern() method (now the pattern tar can be located anywhere)
-  inserted print format for test number 
-  fix how the silkrpc daemon is killed
-  inserted calls of vegeta command directly without script
-  inserted possible configuration of core assigned to vegeta test
- modified geth_home_dir into geth_build_dir
- inserted possible configuration of the silkrpc build
-  input param managed with getopt()

### CODE REVIEW

File: tests/perf/run_perf_tests.py [hunk 1]
```diff
@@ -7,69 +7,80 @@
...
 import sys
+import getopt
 from datetime import datetime
 
 DEFAULT_TEST_SEQUENCE = "50:30,200:30,200:60,400:30"
 DEFAULT_REPETITIONS = 10
-DEFAULT_VEGETA_PATTERN_TAR_FILE = ""
-DEFAULT_DAEMON_ON_CORE = "-"
-DEFAULT_GETH_HOME_DIR = "../../../turbo-geth/"
+DEFAULT_VEGETA_PATTERN_TAR_FILE = "turbo_geth_stress_test_001.tar"
```
Comment (canepat) [inline tests/perf/run_perf_tests.py:15]: I would change this default to something like `./vegeta/turbo_geth_stress_test_001.tar` and put the file under Git control

File: tests/perf/run_perf_tests.py [hunk 2]
```diff
@@ -7,69 +7,80 @@
...
+    print("Usage: " + argv[0] + " -h -p vegetaPatternTarFile -c daemonOnCore  -t turboGethAddress -g turboGethHomeDir -s silkrpcBuildDir -r testRepetitions - t testSequence")
+    print("")
+    print("Launch an automated performance test sequence on Silkrpc and RPCDaemon using Vegeta")
+    print("")
+    print("-h                      print this help")
+    print("-p vegetaPatternTarFile path to the request file for Vegeta attack")
+    print("-c daemonVegetaOnCore   cpu list in taskset format for daemon & vegeta (e.g. 0-1:2-3 or 0-2:3-4 or 0,2:3,4...) [default: " + DEFAULT_DAEMON_VEGETA_ON_CORE +"]")
+    print("-t turboGethAddress     address of TG Core component as <address>:<port> (e.g. localhost:9090)           [default: " + DEFAULT_TG_ADDRESS + "]")
+    print("-g turboGethHomeDir     path to TG home folder (e.g. ../../../turbo-geth/)                               [default: " + DEFAULT_GETH_BUILD_DIR + "]")
+    print("-s silkrpcBuildDir      path to build home folder (e.g. ../../build_gcc_release/)                        [default: " + DEFAULT_SILKRPC_BUILD_DIR + "]")
```
Comment (canepat) [inline tests/perf/run_perf_tests.py:37]: `path to build home folder` -> `path to Silkrpc build folder`

File: tests/perf/run_perf_tests.py [hunk 3]
```diff
@@ -7,69 +7,80 @@
...
+                self.geth_builddir = optarg
+            elif option == "-s":
+                self.silkrpc_build_dir = optarg
+            elif option == "-r":
+                self.repetitions = int(optarg)
+            elif option == "-t":
+                self.test_sequence = optarg
+            else:
+                usage(argv)
+                sys.exit(-1)
```
Comment (canepat) [inline tests/perf/run_perf_tests.py:83]: Move lines 63-83 within `try` block after line 57 for better program flow separation

**Review by canepat — CHANGES_REQUESTED** [review]

File: tests/perf/run_perf_tests.py (hunk 1 above)
Comment (Paolob69) [reply tests/perf/run_perf_tests.py:15]: ok

File: tests/perf/run_perf_tests.py (hunk 2 above)
Comment (Paolob69) [reply tests/perf/run_perf_tests.py:37]: ok

File: tests/perf/run_perf_tests.py [hunk 4]
```diff
@@ -7,69 +7,84 @@
...
+                    self.geth_builddir = optarg
+                elif option == "-s":
+                    self.silkrpc_build_dir = optarg
+                elif option == "-r":
+                    self.repetitions = int(optarg)
+                elif option == "-t":
+                    self.test_sequence = optarg
+                else:
+                    usage(argv)
+                    sys.exit(-1)
```
Comment (canepat) [inline tests/perf/run_perf_tests.py:84]: Lines from 64 to 84 should be moved into the `try` block at line 55 after line 56

File: tests/perf/run_perf_tests.py [hunk 5]
```diff
@@ -7,69 +7,84 @@
...
+        except:
+            usage(argv)
+            sys.exit(-1)
```
Comment (canepat) [inline tests/perf/run_perf_tests.py:87]: This `except` block should be removed, hence the linter error `W0702: No exception type(s) specified (bare-except)` will vanish

**Review by canepat — CHANGES_REQUESTED** [review]

File: tests/perf/run_perf_tests.py (hunk 5 above)
Comment (Paolob69) [reply tests/perf/run_perf_tests.py:87]: ok

File: tests/perf/run_perf_tests.py (hunk 4 above)
Comment (Paolob69) [reply tests/perf/run_perf_tests.py:84]: ok

File: tests/perf/run_perf_tests.py [hunk 6]
```diff
@@ -7,69 +7,84 @@
...
-            print("testSequence             list of query-per-sec and duration tests as <qps1>:<t1>,... (e.g. 200:30,400:10) [default: " + DEFAULT_TEST_SEQUENCE + "]")
+        try:
+            opts, _ = getopt.getopt(argv[1:], "vp:c:a:g:s:r:t:")
+        except getopt.GetoptError as err:
+            # print help information and exit:
+            print(err)
+            usage(argv)
             sys.exit(-1)
 
+        try:
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/run_perf_tests.py:63]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**